man/man1/man1.mk

```diff
@@ -1,4 +1,8 @@
-man1_MANS += man/man1/pkcsconf.1 man/man1/pkcsicsf.1
+man1_MANS += man/man1/pkcsconf.1
+
+if ENABLE_ICSFTOK
```

Comment:
I get 

```
make[1]: *** No rule to make target 'man/man1/pkcsicsf.1', needed by 'all-am'.  Stop.
make[1]: Leaving directory '/mnt/c/Temp/opencryptoki-AIX'
make: *** [Makefile:4216: all] Error 2
```

when building on x86 linux..... Don't really understand why.

Reply:
This wasn't the case before your changes. So I guess it has something to do with this commit or the changes for check_and_enable() in the previous commit.

Reply:
When I revert the hunk with check_and_enable() in the previous commit, it works, and I do see that the man page files aee being generated freshly during make if the were previously cleaned by 'make clean'.:

`cd . && /bin/sh ./config.status man/man1/pkcsicsf.1`

Reply:
This change in `man1.mk` fixes this behaviour:
```diff
 EXTRA_DIST += $(man1_MANS)
-CLEANFILES += man/man1/*.1
+CLEANFILES += $(man1_MANS)
```

Without the hunk you reverted, it works fine because pkcsicsf.1 was always configured, so it would always be available. Ever since it was conditionally configured, it wouldn't be available because the file would be cleared regardless of whether it was configured.

This appears to be work fine in my limited testing, checking with a full rebuild now.

Reply:
Ah yes, this makes a lot of sense (and works in my tests). Please include this change into your commit as well.

There is the same in the make files in man5, man7, and man8. Should not really matter there at the moment, but while you are at it, please fix it there as well.

Reply:
hm, no, still fails....
```
./bootstap.sh
./configure 
make clean
make -> fails
```

Reply:
Ah, same. Is it because of `pkcshsm_mk_change.1` now?

Oddly it worked fine when I first tried it in a dirty build, it fails with a clean configure+build.

Reply:
I think we can get rid of the `CLEANFILES` assignment altogether. From the [automake manual](https://www.gnu.org/software/automake/manual/html_node/Scripts.html)-
```
Another common way to build scripts is to process them from configure with AC_CONFIG_FILES. In this situation Automake knows which files should be cleaned and distributed, and what the rebuild rules should look like. 
[..]
Automake infers them from AC_CONFIG_FILES (see [Configuration requirements](https://www.gnu.org/software/automake/manual/html_node/Requirements.html)). CLEANFILES is still useful, because by default Automake will clean targets of AC_CONFIG_FILES in distclean, not clean.
```
Which means any changes to the `.in` file should automatically be picked up by make and reconfigured appropriately. Early tests suggest this avoids the issue as well. Trying to find out the other semantics of this variable.

Reply:
And that makes sense - if `make` is run without running `configure`, that means the `AC_CONFIG_FILES` might not be re-run, which would explain why the files are missing.

If `CLEANFILES` is updated to include the configured man pages, that means those files will be removed - and without a `configure` run again the file will never be generated.

Reply:
Right, good point. Removing the CLEANFILES from the man make files does solve the problem in my tests.

So I guess we should remove the CLEANFILES from all the man-make files then.